app/search_builders/collection_search_builder.rb

```diff
@@ -9,9 +11,9 @@ def include_item_ids(solr_parameters)
     solr_parameters[:fq] << "hasCollectionMember_ssim:#{item.id}"
   end
 
-  def initialize(item, processor_chain, scope)
-    super(processor_chain, scope)
-    @item = item
+  def include_contained_files(solr_parameters)
+    solr_parameters[:fq] ||= []
+    solr_parameters[:fq] << "{!join from=generic_files_tesim to=id}{!join from=hasCollectionMember_ssim to=id}id:#{collection.id}"
```

Comment:
I'm not that well versed on Solr joins. Could someone like @atz or @jcoyne or @cbeer comment on this line?